compiler/rustc_codegen_ssa/src/back/symbol_export.rs

```diff
@@ -110,7 +110,8 @@ fn reachable_non_generics_provider(tcx: TyCtxt<'_>, _: LocalCrate) -> DefIdMap<S
             // We have to preserve the symbols of the built-in functions during LTO.
             let is_builtin_fn = is_compiler_builtins
                 && symbol_export_level(tcx, def_id.to_def_id())
-                    .is_below_threshold(SymbolExportLevel::C);
+                    .is_below_threshold(SymbolExportLevel::C)
+                && name.contains("compiler_builtins");
```

Comment:
I don't think we should check the name here. The compiler-builtins crate may have a different name. Maybe check for `#[no_mangle]` instead?

Reply:
Yeah, it's better.